### ISSUE TITLE
SC2034 should not trigger for a nameref variable in an assignment only

### ISSUE THREAD

**Issue by leagris** [koalaman/shellcheck#2071]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

declare -n ref=_

for ref in foo bar baz
do ref=hello
done
```

#### Here's what shellcheck currently says:
```
Line 6	SC2034: ref appears unused. Verify use (or export if used externally).
```
#### Here's what I wanted or expected to see:

`SC2034: ref appears unused` should not be triggered because:
- The nameref variable **is effectively used** to assign a value to its successive referees (`foo`, `bar` and `baz` variable names).

**Comment by matthewpersico** [comment]
Furthemore, using a declare -n and NOT assigning to it should be warning, something like `Using nameref foo, but not assigning to it. Consider using traditional pass-by-value instead.`

**Comment by matthewpersico** [comment]
Not sure if this is a specific case of the given example or a more general case but this also throws SC2034:
````
declare -a alist=(a b c)
declare -A ahash=([x]=1 [y]=2 [z]=3)
myfunc eek oof alist ahash
```

**Comment by matthewpersico** [comment]
Confused. I could have sworn that 0.8.0 was supposed to fix this. Still an issue in 0.8.0.

**Comment by matthewpersico** [comment]
> Confused. I could have sworn that 0.8.0 was supposed to fix this. Still an issue in 0.8.0.

Withdrawn. I *is* fixed. I had an error in my code.

**Comment by rbairwell** [comment]
Appears to be a duplicate of https://github.com/koalaman/shellcheck/issues/817 (yep, I came to report the same annoyance...)

**Comment by sergiobaro** [comment]
I'm having this issue with version 0.9.0

**Comment by l1mey112** [comment]
Having this issue with 0.11.0

<img width="1118" height="405" alt="Image" src="https://github.com/user-attachments/assets/5d86e305-048b-4695-8759-fbb357b6aa24" />
